mirdata/datasets/baf.py

```diff
@@ -0,0 +1,436 @@
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            metadata=self._track_metadata,
+        )
+
+
+# no decorator here because of https://github.com/librosa/librosa/issues/1267
+def load_audio(fpath: str) -> Tuple[np.ndarray, float]:
+    """Load a baf audio file.
+
+    Args:
+        fpath (str): path to audio file
+
+    Returns:
+        * np.ndarray - the mono audio signal
+        * float - The sample rate of the audio file
+
+    """
+    return librosa.load(fpath, sr=8000, mono=True)
+
+
+def load_matches(track_metadata: dict) -> Optional[EventDataExtended]:
+    """Load the matches corresponding to a query track.
+
+    Args:
+        track_metadata (dict): track's metadata
+
+    Returns:
+        Optional[EventDataExtended]: Track's annotations in EvendDataExtended format
+    """
+    # intervals_list = deque()  # linked list
+    intervals_list = []
+    events = []
+    tags = []
+    if track_metadata["annotations"] == []:
+        return None
+    else:
+        for ann in track_metadata["annotations"]:
+            intervals_list.append(
+                [round(ann["query_start"], 3), round(ann["query_end"], 3)]
+            )
+            events.append(ann["reference"])
+            tags.append(ann["tag"])
+        intervals = np.array(
+            intervals_list, dtype=float
+        )  # more efficient than appending to np.array
+        return EventDataExtended(
+            intervals=intervals,
+            interval_unit="s",
+            events=events,
+            event_unit="open",
+            tags=tags,
+            tag_unit="open",
+        )
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The BAF dataset
+    """
+
+    def __init__(self, data_home=None, version="default"):
+        super().__init__(
+            data_home,
+            version,
+            name="baf",
+            track_class=Track,
+            bibtex=BIBTEX,
+            indexes=INDEXES,
+            remotes=REMOTES,
+            download_info=DOWNLOAD_INFO,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.cached_property
+    def _metadata(self):
+        """Ingest dataset metadata"""
+        metadata_path = os.path.join(self.data_home, "queries_info.csv")
+        xannotations_path = os.path.join(self.data_home, "cross_annotations.csv")
+        try:
+            metadata_df = pd.read_csv(metadata_path)
+        except FileNotFoundError as not_found:
+            raise FileNotFoundError(
+                FILENOTFOUND_MSG.safe_substitute(fname=not_found.filename)
+            )
+        try:
+            xannotations_df = pd.read_csv(xannotations_path)
+        except FileNotFoundError as not_found:
+            raise FileNotFoundError(
```

Comment:
This is not covered by the tests. I guess you could do something in the tests like:
```
with pytest.raises(FileNotFoundError):
    feat = baf._metadata(/a/fake/path)
```